compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1425,19 +1425,16 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                         this.arena.alloc_from_iter(bounds.iter().filter_map(|bound| match bound {
                             GenericBound::Trait(
                                 ty,
-                                modifier @ (TraitBoundModifier::None
```

Comment:
Negligible behavior change in the error path: We now `lower_poly_trait_ref` for `~const !` (master: `TBM::MaybeConstNegative`) instead of mapping it to `None`. We now only map `?` & `~const ?` to `None` which seems more logical. In any case, `~const !` gets rejected in AST validation, shouldn't have any user-facing effect.